pkg/encrypt/encrypt.go

```diff
@@ -0,0 +1,72 @@
+package encrypt
+
+import (
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"golang.org/x/crypto/chacha20poly1305"
+)
+
+const prefix = "ENC*"
+
+type Cipher interface {
+	Decrypt(string) (string, error)
+	Encrypt(string) (string, error)
+}
+
+var _ Cipher = (*ChaChaCipher)(nil)
+
+type ChaChaCipher struct {
+	aead cipher.AEAD
+}
+
+func New(key []byte) (*ChaChaCipher, error) {
+	i, err := rand.Read(key)
```

Comment:
are you overwriting the key you read from key vault?

Reply:
Yes I do :) hence no review in the comments :dagger: